File: Lecture006 Loops Extra Class/Must Do Patterns in Java/symmetricVoidPattern.java
import java.util.Scanner;

public class symmetricVoidPattern {
    public static void pattern19(int n){
        int InSp = 0;
        for(int i=0;i<n;i++){
            //stars
            for(int j=1;j<=n-i;j++){
                System.out.print("*");
            }
            //spaces
            for(int space = 0; space < InSp; space++){
                System.out.print(" ");
            }
            //stars
                        for(int j=1;j<=n-i;j++){
                System.out.print("*");
            }
            InSp+=2;
            System.out.println();
        }
        InSp = 2 * (n - 1);

    for(int i = 0; i < n; i++){
        // stars
        for(int j = 1; j <= i + 1; j++){
            System.out.print("*");
        }

        // spaces
        for(int space = 0; space < InSp; space++){
            System.out.print(" ");
        }

        // stars
        for(int j = 1; j <= i + 1; j++){
            System.out.print("*");
        }

        InSp -= 2;
        System.out.println();
    }
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        for(int i=0;i<t;i++){
            int n = sc.nextInt();
            pattern19(n);
        }
        sc.close();
    }
}

File: Lecture006 Loops Extra Class/Must Do Patterns in C++/symmetric-voidPattern.cpp
#include<bits/stdc++.h>
using namespace std;

void pattern19(int n){
    // initial space
    int InSp = 0;
    
    for(int i = 0; i < n; i++){
        // stars
        for(int j = 1; j <= n - i; j++){
            cout << "*";
        }

        //spaces
        for(int space = 0; space < InSp; space++){
            cout << " ";
        }

        // stars
        for(int j = 1; j <= n - i; j++){
            cout << "*";
        }

        InSp += 2;
        cout << endl;
    }

    // reset initial space for the lower half
    InSp = 2 * (n - 1);

    for(int i = 0; i < n; i++){
        // stars
        for(int j = 1; j <= i + 1; j++){
            cout << "*";
        }

        // spaces
        for(int space = 0; space < InSp; space++){
            cout << " ";
        }

        // stars
        for(int j = 1; j <= i + 1; j++){
            cout << "*";
        }

        InSp -= 2;
        cout << endl;
    }
}
int main()
{
    int t;
    cin>>t;
    for(int i=0;i<t;i++){
        int n;
        cin>>n;
        pattern19(n);
    }
    return 0;
}